controls/os_spec.rb

```diff
@@ -38,13 +59,19 @@
     it { should exist }
     it { should be_file }
     it { should be_owned_by 'root' }
-    its('group') { should eq 'root' }
     it { should_not be_executable }
     it { should be_writable.by('owner') }
     it { should be_readable.by('owner') }
-    it { should_not be_readable.by('group') }
     it { should_not be_readable.by('other') }
   end
+  describe file('/etc/shadow'), :if => os.family == 'redhat' do
+    its('group') { should eq 'root' }
+    it { should_not be_readable.by('group') }
+  end
+  describe file('/etc/shadow'), :if => os.family == 'debian' do
+    its('group') { should eq 'shadow' }
+    it { should be_readable.by('group') }
```

Comment:
I think it's more my testing experience on Ubuntu...
also see here https://help.ubuntu.com/community/FilePermissions